src/FFmpegUtilities.h

```diff
@@ -126,6 +126,16 @@
 		#define PIX_FMT_YUV444P AV_PIX_FMT_YUV444P
 	#endif
 
+	// Does ffmpeg pixel format contain an alpha channel?
+	inline static const bool ffmpeg_has_alpha(PixelFormat pix_fmt)
+	{
+		if (pix_fmt == AV_PIX_FMT_ARGB || pix_fmt == AV_PIX_FMT_RGBA || pix_fmt == AV_PIX_FMT_ABGR || pix_fmt == AV_PIX_FMT_BGRA || pix_fmt == AV_PIX_FMT_YUVA420P) {
+			return true;
+		} else {
+			return false;
+		}
+	}
```

Comment:
```c++
	inline static const bool ffmpeg_has_alpha(PixelFormat pix_fmt)
	{
		if (pix_fmt == AV_PIX_FMT_ARGB || pix_fmt == AV_PIX_FMT_RGBA || pix_fmt == AV_PIX_FMT_ABGR || pix_fmt == AV_PIX_FMT_BGRA || pix_fmt == AV_PIX_FMT_YUVA420P) {
			return true;
		} else {
			return false;
		}
	}
```

For maintainability this might be better as,
```c++
        inline static const bool ffmpeg_has_alpha(PixelFormat pix_fmt) {
            const AVPixFmtDescriptor *fmt_desc = av_pix_fmt_desc_get(pix_fmt);
            return bool(desc->flags & AV_PIX_FMT_FLAG_ALPHA);
        }
```
That way it'll support any new pixel formats they add in the future, too. Near as I can tell it'd be compatible with every FFmpeg back to at least 2.4, which is already way farther back than we should worry about going.

Reply:
Actually the `static` storage specifier on that may have to come off, if it's going to be taking struct-pointer return values from `av_pix_fmt_desc_get()` and dereferencing them. But an `inline` (non-`static`) function should work fine, I think?

Otherwise, it can always just be added to `FFmpegReader.cpp`.

Reply:
@jonoomph See my previous comment(s), regarding your email question.

Reply:
@jonoomph Oops. Just realized this had a typo in it, it should've been
```c++
        inline static const bool ffmpeg_has_alpha(PixelFormat pix_fmt) {
            const AVPixFmtDescriptor *fmt_desc = av_pix_fmt_desc_get(pix_fmt);
            return bool(fmt_desc->flags & AV_PIX_FMT_FLAG_ALPHA);
        }
```

Reply:
This looks terrific. I'm gonna test it right now! This is exactly what I was searching for, lol. I have a few transparent videos I've been using to test this, so it should be a quick check.